Source/Core/Core/FreelookManager.cpp

```diff
@@ -0,0 +1,223 @@
+  }
+}
+
+void FreelookController::Update()
+{
+  g_freelook_camera.SetControlType(Freelook::g_ActiveConfig.cameraConfig.iControlType);
+
+  if (g_freelook_camera.IsActive())
+  {
+    if (m_move_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.MoveVertical(-g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[1]->GetState<bool>())
+      g_freelook_camera.MoveVertical(g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.MoveHorizontal(g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[3]->GetState<bool>())
+      g_freelook_camera.MoveHorizontal(-g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[4]->GetState<bool>())
+      g_freelook_camera.MoveForward(g_freelook_camera.GetSpeed());
+
+    if (m_move_buttons->controls[5]->GetState<bool>())
+      g_freelook_camera.MoveForward(-g_freelook_camera.GetSpeed());
+
+    if (m_fov_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.IncreaseFovX(g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[1]->GetState<bool>())
+      g_freelook_camera.IncreaseFovX(-1.0f * g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.IncreaseFovY(g_freelook_camera.GetFovStepSize());
+
+    if (m_fov_buttons->controls[3]->GetState<bool>())
+      g_freelook_camera.IncreaseFovY(-1.0f * g_freelook_camera.GetFovStepSize());
+
+    if (m_speed_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.ModifySpeed(1.0f / 1.1f);
+
+    if (m_speed_buttons->controls[1]->GetState<bool>())
+      g_freelook_camera.ModifySpeed(1.1f);
+
+    if (m_speed_buttons->controls[2]->GetState<bool>())
+      g_freelook_camera.ResetSpeed();
+
+    if (m_other_buttons->controls[0]->GetState<bool>())
+      g_freelook_camera.Reset();
+  }
+
+  if (m_other_buttons->controls[1]->GetState<bool>())
+  {
+    if (!m_toggle_pressed_last)
+    {
+      const Freelook::ControlType current_control_type =
+          Freelook::g_ActiveConfig.cameraConfig.iControlType;
+      if (current_control_type != Freelook::ControlType::None)
+        g_freelook_camera.ToggleEnabled();
+      OSD::AddMessage(
+          StringFromFormat("Freelook: %s", g_freelook_camera.IsActive() ? "Enabled" : "Disabled"));
+    }
+    m_toggle_pressed_last = true;
+  }
+  else
+  {
+    m_toggle_pressed_last = false;
+  }
+}
+
+namespace Freelook
+{
+static InputConfig s_config("FreelookController", _trans("Freelook"), "FreelookController");
+InputConfig* GetConfig()
+{
+  return &s_config;
+}
+
+void Shutdown()
+{
+  s_config.UnregisterHotplugCallback();
+
+  s_config.ClearControllers();
+}
+
+void Initialize()
+{
+  if (s_config.ControllersNeedToBeCreated())
+  {
+    s_config.CreateController<FreelookController>(0);
+  }
+
+  s_config.RegisterHotplugCallback();
+
+  Freelook::g_Config.Refresh();
+
+  // Load the saved controller config
```

Comment:
This comment seems a bit pointless as it is right now. We can see it loads from config (and can probably guess that it is the controller config), but I don't see a particular reason why there needs to be a comment pointing out that "we load the saved config".

Is there anything else this comment should probably tells us? And if so, does it also apply to the one in `LoadConfig`?

Reply:
This structure was copied from Gamecube.  I'm pretty sure the comment existed there.  I will remove it